app/src/main/java/com/example/parkingsystemkotlin/mvp/presenter/ReservationParkingPresenter.kt

```diff
@@ -14,6 +22,56 @@ class ReservationParkingPresenter(private val view: ReservationParkingContract.R
     }
 
     override fun onButtonPressedSaveReserver() {
-        view.returnToMainActivity()
+        model.setReservationLotCode(
+            getIntegerUserData(view.getParkingLot()),
+            getIntegerUserData(view.getParkingCode())
+        )
+        val reservation: Reservation? = model.getReservation()
+        when (model.reservationVerify(reservation)) {
+            EnumReservationVerify.MISSING_DATESTART -> view.toastShowMissingDate()
+            EnumReservationVerify.MISSING_DATEEND -> view.toastShowMissingDate()
+            EnumReservationVerify.MISSING_LOT -> view.toastShowMissingLot()
+            EnumReservationVerify.MISSING_CODE -> view.toastShowMissingCode()
+            EnumReservationVerify.RESERVATION_OVERLAP -> view.toastShowOverlap()
+            EnumReservationVerify.COMPROBATION_OK -> {
+                if (reservation != null) {
+                    view.toastShowReserveOK()
+                    model.addReservation(reservation)
+                    view.toastShowTextData(
+                        reservation.parkingLot,
+                        reservation.userCode,
+                        getStringDate(reservation.startDate),
+                        getStringDate(reservation.endDate)
+                    )
+                    view.returnToMainActivity()
+                }
+            }
+        }
     }
+
+    override fun setReservation(calendarDateTime: Calendar?, startDate: Boolean) {
+        if (startDate) {
+            if (calendarDateTime != null) {
+                model.setStartDate(calendarDateTime)
+            }
+        } else {
+            if (calendarDateTime != null) {
+                model.setEndDate(calendarDateTime)
+            }
+        }
+    }
+
+    private fun getStringDate(calendar: Calendar?): String {
+        val date = calendar?.time
+        val simpleDateFormat = SimpleDateFormat(Constant.FORMAT_DATE)
+        var dateToString: String = Constant.EMPTY_STRING
+        try {
+            dateToString = simpleDateFormat.format(date)
+        } catch (exceptionCalendar: Exception) {
+            exceptionCalendar.printStackTrace()
```

Comment:
here you should show an error message or something

Reply:
Done